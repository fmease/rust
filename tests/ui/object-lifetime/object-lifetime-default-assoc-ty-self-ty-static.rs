// FIXME: Explainer.
//@ known-bug: unknown

trait Outer { type Ty; }
trait Inner {}

impl<'a> Outer for dyn Inner + 'a { type Ty = &'a (); }

// FIXME: Deduce `dyn Inner + 'static` from absence of any bounds on self ty param of trait `Outer`.
fn f<'r>(x: &'r <dyn Inner as Outer>::Ty) { g(x) }
fn g<'r>(x: &'r <dyn Inner + 'static as Outer>::Ty) {}

fn main() {}
