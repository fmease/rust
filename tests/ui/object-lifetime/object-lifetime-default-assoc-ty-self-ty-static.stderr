error[E0228]: the lifetime bound for this object type cannot be deduced from context; please supply an explicit bound
  --> $DIR/object-lifetime-default-assoc-ty-self-ty-static.rs:10:18
   |
LL | fn f<'r>(x: &'r <dyn Inner as Outer>::Ty) { g(x) }
   |                  ^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0228`.
